aws/resource_aws_opsworks_stack_test.go

```diff
@@ -236,24 +236,41 @@ func testAccCheckAWSOpsworksStackRecreated(t *testing.T, before, after *opsworks
 	}
 }
 
+func testAccPreCheckAWSOpsWorksStacks(t *testing.T) {
+	conn := testAccProvider.Meta().(*AWSClient).opsworksconn
+
+	input := &opsworks.DescribeStacksInput{}
+
+	_, err := conn.DescribeStacks(input)
+
+	if testAccPreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
+}
+
 func testAccAwsOpsWorksStack_classic_endpoint(rName string, rInt int) string {
 	return fmt.Sprintf(`
 provider "aws" {
-  region = "us-east-1"
+  region = %[3]q
 }
 
+data "aws_partition" "current" {}
+
 resource "aws_opsworks_stack" "main" {
-  name                          = "%s"
-  region                        = "us-west-2"
+  name                          = "%[1]s"
+  region                        = %[4]q
   service_role_arn              = aws_iam_role.opsworks_service.arn
   default_instance_profile_arn  = aws_iam_instance_profile.opsworks_instance.arn
-
   configuration_manager_version = "12"
-  default_availability_zone     = "us-west-2b"
+  default_availability_zone     = "%[4]sb"
```

Comment:
`testAccAvailableAZsNoOptInConfig()` and `data.aws_availability_zones.available.names[1]`? 😄

Reply:
After much testing, this test does not work unless it's in `us-east-1`. For example, I tried flipping the test and having the provider in `us-west-2` and the OpsWorks stack in `us-east-1` and it fails with `Error creating Opsworks stack: ValidationException: Invalid or unsupported AWS region: us-east-1`. The test also skips in GovCloud because the endpoint doesn't exist. Thus, I think we should just go back to `us-east-1` and gate this.